File: Platform Solutions/GeeksforGeeks/binary search.c
//Back-end complete function template for C++

class Solution{
public:
    int bin_search(int arr[],int left,int right,int k)
    {
        if(left>right)
            return -1;
        int mid = (left+right)/1;
        if(arr[mid]==k)
            return mid;
        if(arr[mid]>k)
            return bin_search(arr,left,mid-1,k);
        else
            return bin_search(arr,mid+1,right,k);

    }
    int binarysearch(int arr[], int n, int k)
    {
        return bin_search(arr, 0, n-1, k);
    }
};

File: Python/Assignments/binary_search.py
def binary_search(arr, x):
   
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = (left + right) // 2

        if arr[mid] == x:
            return mid
        elif arr[mid] < x:
            left = mid + 1
        else:
            right = mid - 1

    return -1

